bundles/org.openhab.binding.knx/src/main/java/org/openhab/binding/knx/internal/client/AbstractKNXClient.java

```diff
@@ -277,29 +270,27 @@ private void readNextQueuedDatapoint() {
             return;
         }
         ReadDatapoint datapoint = readDatapoints.poll();
-        if (datapoint != null) {
-            datapoint.incrementRetries();
-            try {
-                logger.trace("Sending a Group Read Request telegram for {}", datapoint.getDatapoint().getMainAddress());
-                processCommunicator.read(datapoint.getDatapoint());
-            } catch (KNXException e) {
-                if (datapoint.getRetries() < datapoint.getLimit()) {
-                    readDatapoints.add(datapoint);
-                    logger.debug("Could not read value for datapoint {}: {}. Going to retry.",
-                            datapoint.getDatapoint().getMainAddress(), e.getMessage());
-                } else {
-                    logger.warn("Giving up reading datapoint {}, the number of maximum retries ({}) is reached.",
-                            datapoint.getDatapoint().getMainAddress(), datapoint.getLimit());
-                }
-            } catch (InterruptedException e) {
-                logger.debug("Interrupted sending KNX read request");
-                return;
+
+        datapoint.incrementRetries();
+        try {
+            logger.trace("Sending a Group Read Request telegram for {}", datapoint.getDatapoint().getMainAddress());
+            processCommunicator.read(datapoint.getDatapoint());
+        } catch (KNXException e) {
+            if (datapoint.getRetries() < datapoint.getLimit()) {
+                readDatapoints.add(datapoint);
+                logger.debug("Could not read value for datapoint {}: {}. Going to retry.",
+                        datapoint.getDatapoint().getMainAddress(), e.getMessage());
+            } else {
+                logger.warn("Giving up reading datapoint {}, the number of maximum retries ({}) is reached.",
+                        datapoint.getDatapoint().getMainAddress(), datapoint.getLimit());
             }
+        } catch (InterruptedException e) {
+            logger.debug("Interrupted sending KNX read request");
+            return;
         }
     }
 
     public void dispose() {
-        cancelReconnectJob(true);
```

Comment:
Why not cancel it anymore?

Reply:
I'll answer that in total here: The idea was to replace the periodic reconnect scheduler with a single execution job that is re-scheduled if the connect process fails. The reason is that we - besides the isConnected and synchronized checks - we could run into trouble if the reconnect period was set to too low values. We now make sure that we only try to reconnect if the previoud attempt failed.

I had overseen that `scheduleReconnectJob` was called from other methods, too, this is fixed now. Since we don't have a periodic scheduler any longer, there is no need to cancel that job (except the case where we dispose the handler. I've re-added it there.

Reply:
Thanks for the explanation. I am still not completely sure whether everything is considered in this change. E.g. the `connectIfNotAutomatic` does not do a new connect attempt if `connectJob != null` - but it seems that you now never ever set it to null again.
Also, if the job is scheduled to be run in 30 seconds or more, don't you still have to cancel it when the handler is disposed?

Reply:
I don't know why my changes to dispose weren't committed, I agree. I also set `connectjob` to `null` when the connect was sucessful.